frontend/src/pages/integrations/databricks/create.tsx

```diff
@@ -0,0 +1,251 @@
+import {
+  faArrowUpRightFromSquare,
+  faBookOpen,
+  faBugs,
+  faCircleInfo
+} from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import queryString from "query-string";
+
+import { createNotification } from "@app/components/notifications";
+import { useCreateIntegration } from "@app/hooks/api";
+
+import {
+  Button,
+  Card,
+  CardTitle,
+  FormControl,
+  Input,
+  Select,
+  SelectItem
+} from "../../../components/v2";
+import {
+  useGetIntegrationAuthApps,
+  useGetIntegrationAuthById
+} from "../../../hooks/api/integrationAuth";
+import { useGetWorkspaceById } from "../../../hooks/api/workspace";
+
+export default function DatabricksCreateIntegrationPage() {
+  const router = useRouter();
+  const { mutateAsync } = useCreateIntegration();
+
+  const { integrationAuthId } = queryString.parse(router.asPath.split("?")[1]);
+
+  console.log("integrationAuthId", integrationAuthId)
+
+  const { data: workspace } = useGetWorkspaceById(localStorage.getItem("projectData.id") ?? "");
+  const { data: integrationAuth, isLoading: isintegrationAuthLoading } = useGetIntegrationAuthById(
+    (integrationAuthId as string) ?? ""
+  );
+
+  console.log("integrationAuth", integrationAuth)
+  const { data: integrationAuthScopes, isLoading: isIntegrationAuthScopesLoading } =
+    useGetIntegrationAuthApps({
+      integrationAuthId: (integrationAuthId as string) ?? ""
+    });
+
+  console.log("integrationAuthScopes", integrationAuthScopes)
+  const [selectedSourceEnvironment, setSelectedSourceEnvironment] = useState("");
+  const [targetScope, setTargetScope] = useState("");
+  const [secretPath, setSecretPath] = useState("/");
+
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    if (workspace) {
+      setSelectedSourceEnvironment(workspace.environments[0].slug);
+    }
+  }, [workspace]);
+
+  const handleButtonClick = async () => {
+    try {
+      if (!integrationAuth?.id) return;
+
+      if (!targetScope) {
+        createNotification({
+          type: "error",
+          text: "Please select a scope"
+        });
+        setIsLoading(false);
+        return;
+      }
+
+      setIsLoading(true);
+
+      const selectedScope = integrationAuthScopes?.find(
+        (integrationAuthScope) => integrationAuthScope.name === targetScope
+      );
+
+      if (!selectedScope) {
+        createNotification({
+          type: "error",
+          text: "Invalid scope selected"
+        });
+        setIsLoading(false);
+        return;
+      }
+
+      await mutateAsync({
+        integrationAuthId: integrationAuth?.id,
+        isActive: true,
+        app: selectedScope.name, // scope name
+        sourceEnvironment: selectedSourceEnvironment,
+        secretPath
+      });
+
+      setIsLoading(false);
+
+      router.push(`/integrations/${localStorage.getItem("projectData.id")}`);
+    } catch (err) {
+      console.error(err);
+    }
```

Comment:
same as above - suggest moving `setIsLoading(false);` after `catch` or in `finally` or using the mutation hook `isLoading`